src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -39,9 +39,14 @@ public String getToken() {
     // Called from other synchronized code. Do not call into other synchronized code or you'll
     // risk deadlock
     public synchronized void setDistinctId(String distinctId) {
-        mUnseenSurveys.clear();
-        mUnseenNotifications.clear();
-        mDistinctId = distinctId;
+        if (mDistinctId.equals(distinctId)) {
+            return; // the user is the same, so no new response is needed
```

Comment:
Not a big fan of return statements at the beginning like this. I'd either do:
```
if (something)
   return

code
```

or

```
if (!something) {
    code
}
```